### ISSUE TITLE
Can't correctly check Decommissioned Node

### ISSUE THREAD

**Issue by allamand** [Orange-OpenSource/cassandra-k8s-operator#21]
In some cases when Decommissioning a Node CassKop may not be able to find that the Node is decommissionned, we get this error:

```
DEBU[0317] Creating Jolokia connection                   host=cassandra-demo-dc2-rack1-0.cassandra-demo-dc2 namespace=cassandra-demo port=8778 secretRef="{jolokia-auth}"
DEBU[0317] Using Secret for Jolokia connection           host=cassandra-demo-dc2-rack1-0.cassandra-demo-dc2 namespace=cassandra-demo port=8778 secretRef="{jolokia-auth}"
DEBU[0317] Node decommission ended                       cluster=cassandra-demo pod=cassandra-demo-dc2-rack1-0 rack=dc2-rack1
DEBU[0317] Node decommission failed                      cluster=cassandra-demo err="Cannot decommission: java.lang.UnsupportedOperationException : local node is not a member of the token ring yet" pod=cassandra-demo-dc2-rack1-0 rack=dc2-rack1

```

If we check logs for corresponding Cassandra node:

```
INFO  [jolokia-3] 2019-06-19 09:49:38,509 StorageService.java:4029 - Announcing that I have left the ring for 30000ms
INFO  [jolokia-4] 2019-06-19 09:50:01,702 Server.java:176 - Stop listening for CQL clients
WARN  [jolokia-4] 2019-06-19 09:50:01,703 Gossiper.java:1559 - No local state, state is in silent shutdown, or node hasn't joined, not announcing shutdown
INFO  [jolokia-4] 2019-06-19 09:50:01,704 MessagingService.java:981 - Waiting for messaging service to quiesce
INFO  [ACCEPT-cassandra-demo-dc2-rack1-0.cassandra-demo-dc2.cassandra-demo.svc.cluster.local/10.40.0.26] 2019-06-19 09:50:01,709 MessagingService.java:1336 - MessagingService has terminated the accept() thread
INFO  [jolokia-4] 2019-06-19 09:50:01,832 StorageService.java:1483 - DECOMMISSIONED
WARN  [jolokia-0] 2019-06-19 09:50:02,002 Gossiper.java:1559 - No local state, state is in silent shutdown, or node hasn't joined, not announcing shutdown
INFO  [jolokia-0] 2019-06-19 09:50:02,005 MessagingService.java:981 - Waiting for messaging service to quiesce
INFO  [jolokia-0] 2019-06-19 09:50:02,089 StorageService.java:1483 - DECOMMISSIONED
WARN  [jolokia-1] 2019-06-19 09:50:03,327 Gossiper.java:1559 - No local state, state is in silent shutdown, or node hasn't joined, not announcing shutdown
INFO  [jolokia-1] 2019-06-19 09:50:03,328 MessagingService.java:981 - Waiting for messaging service to quiesce
INFO  [jolokia-1] 2019-06-19 09:50:03,402 StorageService.java:1483 - DECOMMISSIONED
WARN  [jolokia-2] 2019-06-19 09:50:04,324 Gossiper.java:1559 - No local state, state is in silent shutdown, or node hasn't joined, not announcing shutdown
INFO  [jolokia-2] 2019-06-19 09:50:04,325 MessagingService.java:981 - Waiting for messaging service to quiesce
INFO  [jolokia-2] 2019-06-19 09:50:04,360 StorageService.java:1483 - DECOMMISSIONED
WARN  [jolokia-3] 2019-06-19 09:50:08,510 Gossiper.java:1559 - No local state, state is in silent shutdown, or node hasn't joined, not announcing shutdown
INFO  [jolokia-3] 2019-06-19 09:50:08,511 MessagingService.java:981 - Waiting for messaging service to quiesce
INFO  [jolokia-3] 2019-06-19 09:50:08,541 StorageService.java:1483 - DECOMMISSIONED
```

In previous logs, we see that we have lost temporarylly the connection with the node

```
INFO[0164] [cassandra-demo][dc2-rack1]: Scaling Cluster : Ask 0 and have 1 --> ScaleDown 
INFO[0164] ScaleDown detected, we launch decommission    cluster=cassandra-demo pod=cassandra-demo-dc2-rack1-0 rack=dc2-rack1
DEBU[0164] Decommissioning cassandra node                cluster=cassandra-demo pod=cassandra-demo-dc2-rack1-0 rack=dc2-rack1
DEBU[0164] Waiting Rack to be running before continuing, we break ReconcileRack Without Updating Statefulset  cluster=cassandra-demo dc-rack=dc2-rack1 err="<nil>"
DEBU[0164] Node decommission starts                      cluster=cassandra-demo pod=cassandra-demo-dc2-rack1-0 rack=dc2-rack1
DEBU[0164] Creating Jolokia connection                   host=cassandra-demo-dc2-rack1-0.cassandra-demo-dc2 namespace=cassandra-demo port=8778 secretRef="{jolokia-auth}"
DEBU[0164] Using Secret for Jolokia connection           host=cassandra-demo-dc2-rack1-0.cassandra-demo-dc2 namespace=cassandra-demo port=8778 secretRef="{jolokia-auth}"
Request Could not be sendDEBU[0164] Node decommission ended                       cluster=cassandra-demo pod=cassandra-demo-dc2-rack1-0 rack=dc2-rack1
DEBU[0164] Node decommission failed                      cluster=cassandra-demo err="Cannot decommission: HTTP Request Failed: Post http://cassandra-demo-dc2-rack1-0.cassandra-demo-dc2:8778/jolokia/: dial tcp: lookup cassandra-demo-dc2-rack1-0.cassandra-demo-dc2: no such host" pod=cassandra-demo-dc2-rack1-0 rack=dc2-rack1
DEBU[0164] Statefulsets Are Equal: No Update             cluster=cassandra-demo dc-rack=dc1-rack1
DEBU[0165] Statefulsets Are Equal: No Update             cluster=cassandra-demo dc-rack=dc1-rack2
DEBU[0165] Statefulsets Are Equal: No Update             cluster=cassandra-demo dc-rack=dc1-rack3
INFO[0165] [cassandra-demo][dc2-rack1]: StatefulSet(ScaleDown) Replicas Number Not OK: 1 on 1, ready[0] 
INFO[0165] [cassandra-demo][dc2-rack1]: Scaling Cluster : Ask 0 and have 1 --> ScaleDown 
INFO[0165] ScaleDown detected, we launch decommission    cluster=cassandra-demo pod=cassandra-demo-dc2-rack1-0 rack=dc2-rack1
```

**Comment by cscetbon** [comment]
Any way to reproduce that issue ? 

I wonder if 2 decommissions operations are running at once or if there is a timing issue `No local state, state is in silent shutdown, or node hasn't joined, not announcing shutdown`

**Comment by cscetbon** [comment]
@allamand do you confirm it was a false positive ?

**Comment by allamand** [comment]
I was updating the state machine which brings to start several decommission in //.
I wonder If we could change a little the test so that we can know that this node was decommissioned?

**Comment by cscetbon** [comment]
We can check if the node is the ring before trying to decommission it and flag the operation as a success if it's not ?

**Comment by allamand** [comment]
I think that can be a good solution

**Comment by cscetbon** [comment]
see https://github.com/Orange-OpenSource/cassandra-k8s-operator/pull/39

**Comment by cscetbon** [comment]
@allamand you can close that issue.

**Comment by allamand** [comment]
Yes Fix in #39
